src/requests/adapters.py

```diff
@@ -98,25 +115,26 @@ def _urllib3_request_context(
     parsed_request_url = urlparse(request.url)
     scheme = parsed_request_url.scheme.lower()
     port = parsed_request_url.port
-
-    # Determine if we have and should use our default SSLContext
-    # to optimize performance on standard requests.
     poolmanager_kwargs = getattr(poolmanager, "connection_pool_kw", {})
-    has_poolmanager_ssl_context = poolmanager_kwargs.get("ssl_context")
-    should_use_default_ssl_context = (
-        _preloaded_ssl_context is not None and not has_poolmanager_ssl_context
-    )
 
     cert_reqs = "CERT_REQUIRED"
+    cert_loc = None
     if verify is False:
         cert_reqs = "CERT_NONE"
-    elif verify is True and should_use_default_ssl_context:
+    elif _should_use_default_context(verify, client_cert, poolmanager_kwargs):
         pool_kwargs["ssl_context"] = _preloaded_ssl_context
+    elif verify is True:
+        # Set default ca cert location if none provided
+        cert_loc = extract_zipped_paths(DEFAULT_CA_BUNDLE_PATH)
```

Comment:
So this is what fixes #6730 right? Because the custome Context + passing this should make their life better, right?

Reply:
Yeah, this should address #6730. #6667 assumed we would always be using the default context which lead to removing these [two lines](https://github.com/psf/requests/pull/6667/files#diff-a4598bf8444e0feca489e60704c809c0dbfc47b90c4364216c931b497b3312d7L293-L294). That was "ok" because we were always forcing a load on the default context at import time.

Starting in 2.32.3, we've opted out of the default in the failure cases reported between 2.32.0-2.32.2. These are cases we're taking a custom context from the user but no longer setting the `DEFAULT_CA_BUNDLE` on the connection returned from what was previously `get_connection`. Since we don't know if we have the default SSLContext in `cert_verify`, I ported this into `_urllib3_request_context` to ensure we're passing this as `pool_kwargs` when we've disabled the default.

I validated this against the issue reported in #6726 which was showing this regression after the `cert` fix in 2769cb607d4e696e2fe70802d4246ccc5abd64a8. The other reported issues like #6730 are also passing cleanly now.